src/zjs_ble.c

```diff
@@ -405,28 +395,55 @@ static jerry_value_t zjs_ble_update_value_call_function(const jerry_value_t func
     return zjs_error("updateValueCallback: buffer not found or empty");
 }
 
-static void zjs_ble_subscribe_call_function(struct zjs_callback *cb)
+static void zjs_ble_subscribe_c_callback(void *handle)
 {
-    ble_notify_handle_t *mycb;
-    ble_characteristic_t *chrc;
-    mycb = CONTAINER_OF(cb, ble_notify_handle_t, zjs_cb);
-    chrc = CONTAINER_OF(mycb, ble_characteristic_t, subscribe_cb);
+    ble_characteristic_t *chrc = (ble_characteristic_t *)handle;
+    ble_notify_handle_t *cb = &chrc->subscribe_cb;
 
     jerry_value_t rval;
     jerry_value_t args[2];
 
     args[0] = jerry_create_number(20); // max payload size
-    args[1] = jerry_create_external_function(zjs_ble_update_value_call_function);
-    rval = jerry_call_function(mycb->zjs_cb.js_callback, chrc->chrc_obj, args, 2);
+    args[1] = jerry_create_external_function(zjs_ble_update_value_callback_function);
+    rval = jerry_call_function(cb->js_callback, chrc->chrc_obj, args, 2);
     if (jerry_value_has_error_flag(rval)) {
-        PRINT("zjs_ble_subscribe_call_function: failed to call onSubscribe function\n");
+        PRINT("zjs_ble_subscribe_c_callback: failed to call onSubscribe function\n");
     }
 
-    jerry_release_value(args[0]);
     jerry_release_value(args[1]);
     jerry_release_value(rval);
 }
 
+static void zjs_ble_unsubscribe_c_callback(void *handle)
+{
+    ble_characteristic_t *chrc = (ble_characteristic_t *)handle;
+    ble_notify_handle_t *cb = &chrc->unsubscribe_cb;
+
+    jerry_value_t rval;
+
+    rval = jerry_call_function(cb->js_callback, chrc->chrc_obj, NULL, 0);
+    if (jerry_value_has_error_flag(rval)) {
+        PRINT("zjs_ble_unsubscribe_c_callback: failed to call onUnsubscribe function\n");
+    }
+
+    jerry_release_value(rval);
+}
+
+static void zjs_ble_notify_c_callback(void *handle)
+{
+    ble_characteristic_t *chrc = (ble_characteristic_t *)handle;
+    ble_notify_handle_t *cb = &chrc->notify_cb;
+
+    jerry_value_t rval;
+
+    rval = jerry_call_function(cb->js_callback, chrc->chrc_obj, NULL, 0);
+    if (jerry_value_has_error_flag(rval)) {
+        PRINT("zjs_ble_notify_c_callback: failed to call onNotify function\n");
+    }
+
+    jerry_release_value(rval);
+}
```

Comment:
These two functions are identical, I think, so make a more generic name and use it twice?